lib/pages/search/server_search_view.dart

```diff
@@ -0,0 +1,112 @@
+import 'package:fluffychat/utils/matrix_sdk_extensions/matrix_locals.dart';
+import 'package:fluffychat/utils/string_extension.dart';
+import 'package:fluffychat/widgets/avatar/avatar.dart';
+import 'package:fluffychat/widgets/highlight_text.dart';
+import 'package:fluffychat/widgets/matrix.dart';
+import 'package:matrix/matrix.dart';
+import 'package:fluffychat/pages/chat_list/chat_list_item_title.dart';
+import 'package:flutter/material.dart' hide SearchController;
+import 'package:flutter_gen/gen_l10n/l10n.dart';
+
+class UnencryptedMessagesChatList extends StatelessWidget {
+  final SearchController searchController;
+
+  const UnencryptedMessagesChatList({
+    super.key,
+    required this.searchController,
+  });
+
+  static const _prefixLengthHighlight = 20;
+
+  @override
+  Widget build(BuildContext context) {
+    return SliverToBoxAdapter(
+      child: ValueListenableBuilder(
+        valueListenable:
+            searchController.serverSearchController.serverSearchResultNotifier,
+        builder: ((context, searchResults, child) {
+          if (searchResults == null || searchResults.isEmpty) {
+            return const SizedBox.shrink();
+          }
+
+          return Padding(
+            padding: const EdgeInsets.all(8.0),
+            child: ListView.builder(
+              shrinkWrap: true,
+              physics: const ClampingScrollPhysics(),
+              itemCount: searchResults.length,
+              padding: const EdgeInsets.all(8.0),
+              itemBuilder: ((context, index) {
+                final searchResult = searchResults[index];
+                final roomId = searchResult?.roomId;
+                if (roomId == null) {
+                  return const SizedBox.shrink();
+                }
+                final room = Matrix.of(context).client.getRoomById(roomId);
+                if (room == null || searchResult == null) {
+                  return const SizedBox.shrink();
+                }
+                final searchWord = searchController.textEditingController.text;
+                final event = Event.fromMatrixEvent(searchResult, room);
+                final senderName =
+                    event.senderFromMemoryOrFallback.calcDisplayname(
+                  i18n: MatrixLocals(L10n.of(context)!),
+                );
+                final originServerTs = searchResult.originServerTs;
+                final bodyContent = event
+                    .calcLocalizedBodyFallback(
+                      MatrixLocals(L10n.of(context)!),
+                      hideEdit: true,
+                      hideReply: true,
+                      plaintextBody: true,
+                      removeMarkdown: true,
+                    )
+                    .substringToHighlight(
+                      searchWord,
+                      prefixLength: _prefixLengthHighlight,
+                    );
+                final bodyText = '$senderName: $bodyContent';
+                return Padding(
```

Comment:
too much logic here, please move it to controller

Reply:
done

Reply:
please also move get roomId, room, event to controller

Reply:
can't move room, event to controller, because it depends on `searchResult` and other ones depends on it, remove `roomId` to extension
<img width="662" alt="Screenshot 2023-12-05 at 14 30 01" src="https://github.com/linagora/twake-on-matrix/assets/43041967/9ba5c741-1768-4974-9a18-8d0bd8e6c0e1">